### PR TITLE
Use https instead of http where possible

### PR DESCRIPTION
**Pull request by iblech**
I'm very happy that in my social circle, it's considered slightly inpolite to use http instead of https addresses. :-) Since a friend just directed me to this repository, I thought to have a go at rewriting the few remaining http links in this repository.

* I checked [with an automatic tool](https://github.com/luga-ev/website/blob/master/rewrite-http-to-https.pl), and with manual samples, that this pull request doesn't introduce broken links.
* The mirror `repo3.voidlinux.eu` seems to be down (yields 404) and have not rewritten the http links to it.
* I didn't make any changes to [NEWS](NEWS) since its past entries should probably be considered to be an immutable historic recrod.
* The most important change is probably in [README.md](README.md) for referencing the sha256sums.

(I'm not trying to start a holy war and don't particularly care. Merge this or ignore it as you wish, there are no hard feelings from my side.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,17 +36,17 @@ to handle binary packages and repositories. Some highlights:
...
+  - [GNU make](https://www.gnu.org/software/make/)
+  - [pkg-config](https://www.freedesktop.org/wiki/Software/pkg-config/)
+  - [zlib](https://www.zlib.net)
+  - [openssl < 1.1](https://www.openssl.org) or [libressl](https://www.libressl.org/)
+  - [libarchive >= 3.2.0](https://www.libarchive.org)
 
 and optionally:
 
-  - [graphviz](http://www.graphviz.org) and [doxygen](http://www.doxygen.org)
+  - [graphviz](https://www.graphviz.org) and [doxygen](http://www.doxygen.org)
```
Comment (Vaelatern) [inline README.md]: missed one, I think.

File: README.md [hunk 2]
```diff
@@ -36,17 +36,17 @@ to handle binary packages and repositories. Some highlights:
...
     (--enable-api-docs) to build API documentation.
-  - [atf >= 0.15](http://code.google.com/p/kyua) (--enable-tests) to build the
+  - [atf >= 0.15](https://code.google.com/p/kyua) (--enable-tests) to build the
```
Comment (Vaelatern) [inline README.md]: `https://github.com/jmmv/kyua`

File: README.md [hunk 3]
```diff
@@ -82,13 +82,13 @@ Good luck!
...
-* [x86\_64](http://repo.voidlinux.eu/static/xbps-static-latest.x86_64-musl.tar.xz)
-* [mips32](http://repo.voidlinux.eu/static/xbps-static-latest.mips-musl.tar.xz)
+* [aarch64](https://repo.voidlinux.eu/static/xbps-static-latest.aarch64-musl.tar.xz)
+* [armv6hf](https://repo.voidlinux.eu/static/xbps-static-latest.armv6l-musl.tar.xz)
+* [i686](https://repo.voidlinux.eu/static/xbps-static-latest.i686-musl.tar.xz)
+* [x86\_64](https://repo.voidlinux.eu/static/xbps-static-latest.x86_64-musl.tar.xz)
+* [mips32](https://repo.voidlinux.eu/static/xbps-static-latest.mips-musl.tar.xz)
 
-Check the *sha256* checksums with this file: http://repo.voidlinux.eu/static/sha256sums.txt
+Check the *sha256* checksums with this file: https://repo.voidlinux.eu/static/sha256sums.txt
```
Comment (Vaelatern) [inline README.md]: All of these change from `.eu` to `.org`

**Comment by Vaelatern** [comment]
I'm not sure that we want to do this, @Duncaen ?

File: README.md (hunk 3 above)
Comment (iblech) [reply README.md]: You mean `https://alpha.de.repo.voidlinux.org/static/sha256sums.txt` (and similar?).

File: README.md (hunk 1 above)
Comment (iblech) [reply README.md]: This timed out on my first check, but now it works. Will add this in a second commit.

File: README.md (hunk 2 above)
Comment (iblech) [reply README.md]: Thanks, will change this is in a second.

**Comment by Vaelatern** [comment]
I'd like to see the URLs moderized and corrected, and that means the repo urls and our new domain.

**Comment by the-maldridge** [comment]
Modernize and move to the new domains, but not to https, this breaks certain capabilities we currently depend on during maintenance windows, and for servers that support https your connection is upgraded automatically anyway.

**Comment by iblech** [comment]
> for servers that support https your connection is upgraded automatically anyway.

Sure, but at this point there was already a leak of privacy (the unencrypted request).

**Comment by xtraeme** [comment]
This was applied already, I think it can be closed now.

**Comment by the-maldridge** [comment]
Closing as a modified version was applied.  This PR is so dated anyway it would need to be re-done.

**Comment by iblech** [comment]
Original pull request author here, great to learn that a modified version was applied!

**Pull request closed without merging**